bin/brew

```diff
@@ -89,6 +89,7 @@ then
 
   FILTERED_ENV=()
   for VAR in HOME SHELL PATH TERM LOGNAME USER CI TRAVIS http_proxy \
+             https_proxy ftp_proxy no_proxy \
```

Comment:
What is `no_proxy` used for? Can you move `http_proxy` from the previous line, thanks!

Reply:
`no_proxy` lets you list internal domains that should not use the proxy server. Patch updated.

Reply:
Why would you use this with Homebrew?

Reply:
What if somebody had some private casks in their intranet, for instance?

Reply:
@barreyra Which of these variables do you personally need for Homebrew to function? I don't want to whitelist everything that could be useful but just things that individuals need.

Reply:
I don't personally need no_proxy, but I think all others are likely needed. It just depends on what protocol is required to retrieve a given dependency (http/https/ftp).

Reply:
@barreyra I'd suggest just going for `HTTPS_PROXY` (https://github.com/Homebrew/brew/issues/3500) and, if you personally use it, `FTP_PROXY` for now (both uppercase). If you also use `ALL_PROXY` or `NO_PROXY` they can be added too (but only if you personally need them).

Reply:
So *not* setting the http proxy?

What's the difference between the lower and uppercase versions of the variables? I've always used the lowercase versions.

Reply:
Answering my own question, from the curl manpage:

>The  environment  variables  can  be  specified  in lower case or upper case. The lower case version has precedence. http_proxy is an exception as it is only available in lower case.

Why do you prefer the uppercase version?

Reply:
> Why do you prefer the uppercase version?

That's the version that's documented in the man page if you're copy-pasting.

Reply:
I'm fine going with this for now. I had added no_proxy because it was one of the asks in #3500